tests/test_comfy_api.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+from comfy_api_simplified.comfy_api_wrapper import ComfyApiWrapper
+from comfy_api_simplified.comfy_workflow_wrapper import ComfyWorkflowWrapper
+from PIL import Image, ImageChops
+import pytest_asyncio
+
+import io
+import os
+
+
+from .conftest import test_data_dir
+
+
+def compare_images(img1, img2):
+    return ImageChops.difference(img1, img2).getbbox() is None
+
+
+def load_image_from_test_data(image_name):
+    path = os.path.join(test_data_dir, image_name)
+    return Image.open(path)
+
+
+def test_upload_image(api_wrapper: ComfyApiWrapper):
+    path = os.path.join(test_data_dir, "input.png")
+    assert os.path.exists(path)
+    try:
+        response = api_wrapper.upload_image(path)
+    except:
+        pytest.fail("Unexpected error")
+
+    assert response
+
+
+def test_queue_prompt(
+    api_wrapper: ComfyApiWrapper, workflow_wrapper: ComfyWorkflowWrapper
+):
+    api_wrapper.upload_image(os.path.join(test_data_dir, "input.png"))
+
+    response = api_wrapper.queue_prompt(workflow_wrapper)
+    prompt_id = response["prompt_id"]
+    assert api_wrapper.get_queue_size_before(prompt_id) == 0
+
+
+@pytest.mark.asyncio
+async def test_get_history(
+    api_wrapper: ComfyApiWrapper, workflow_wrapper: ComfyWorkflowWrapper
+):
+    api_wrapper.upload_image(os.path.join(test_data_dir, "input.png"))
+
+    prompt_id = await api_wrapper.queue_prompt_and_wait(workflow_wrapper)
+
+    try:
+        response = api_wrapper.get_history(prompt_id)
+    except:
+        pytest.fail("Unexpected error")
+
+    assert response
+
+
+def test_negative_set_node_param(workflow_wrapper: ComfyWorkflowWrapper):
+    try:
+        workflow_wrapper.set_node_param(
+            "a node that does not exist",
+            "image",
+            "",
+        )
+        pytest.fail("Unexpected success")
+    except:
+        pass
+
+
+@pytest.mark.asyncio
+async def test_get_image(
+    api_wrapper: ComfyApiWrapper, workflow_wrapper: ComfyWorkflowWrapper
+):
+    expected_image = load_image_from_test_data("output.png")
+
+    image_metadata = api_wrapper.upload_image(os.path.join(test_data_dir, "input.png"))
+
+    workflow_wrapper.set_node_param(
+        "Load Image",
+        "image",
+        f"{image_metadata['subfolder']}/{image_metadata['name']}",
+    )
+
+    prompt_id = await api_wrapper.queue_prompt_and_wait(workflow_wrapper)
+    response = api_wrapper.get_history(prompt_id)
+
+    print(response[prompt_id]["outputs"]["9"]["images"][0])
+    output = response[prompt_id]["outputs"]["9"]["images"][0]
+    filename = output["filename"]
+    subfolder = output["subfolder"]
+    type = output["type"]
+    image = api_wrapper.get_image(filename, subfolder, type)
+
+    returned_image = Image.open(io.BytesIO(image))
+
+    assert compare_images(
```

Comment:
Somehow your output.png is not the same as my result of the generation during this test, even if by naked eye they look perfectly similar. There could be some issues with different GPU hardware, but I am not sure. So this test is not very reliable as it is now.
I still think it is a good idea for a test, but I suggest to make it simpler:
1. Generate an image using text2image workflow
2. Run  "get_image" and check that the pixel info is not empty and dimentions are 512x512.

If you do not have time to modify it, you can comment this test for now, I will merge it and rewrite later

Reply:
Yeah, it's unfortunate that the results aren't completely reproducible. I modified the check function to look for high similarity instead of exact identity. Also, I like the idea of checking for similarity because it ensures that the input image was correctly used.